modules/generator/registry/interface.go

```diff
@@ -17,6 +17,8 @@ type Counter interface {
 // Histogram
 // https://prometheus.io/docs/concepts/metric_types/#histogram
 type Histogram interface {
+	metric
```

Comment:
it seems like counter and gauge already support this interface as well. should we add to all 3 for clarity?

Reply:
Good catch.  Probably yes, which I think prometheus does this already in a similar way.